pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2818,43 +2839,67 @@ int tracepoint__raw_syscalls__sys_enter(struct bpf_raw_tracepoint_args *ctx)
         sys->id = *id_64;
     }
 
-    // TODO: make this a tailcall (userland might place it if option is given)
-    if (data.config->options & OPT_TRANSLATE_FD_FILEPATH && should_submit(sys->id, data.config) &&
-        has_syscall_fd_arg(sys->id)) {
-        // Process filepath related to fd argument
-        uint fd_num = get_syscall_fd_num_from_arg(sys->id, &sys->args);
-        struct file *file = get_struct_file_from_fd(fd_num);
-
-        if (file) {
-            fd_arg_task_t fd_arg_task = {
-                .pid = data.context.task.pid,
-                .tid = data.context.task.tid,
-                .fd = fd_num,
-            };
+    // exit, exit_group and rt_sigreturn syscalls don't return
+    if (sys->id != SYSCALL_EXIT && sys->id != SYSCALL_EXIT_GROUP &&
+        sys->id != SYSCALL_RT_SIGRETURN) {
+        sys->ts = bpf_ktime_get_ns();
+        task_info->syscall_traced = true;
+    }
 
-            fd_arg_path_t fd_arg_path = {};
-            void *file_path = get_path_str(GET_FIELD_ADDR(file->f_path));
+    bool submitEnter = should_submit(RAW_SYS_ENTER, config);
+    bool submitSyscall = should_submit(sys->id, config);
 
-            bpf_probe_read_str(&fd_arg_path.path, sizeof(fd_arg_path.path), file_path);
-            bpf_map_update_elem(&fd_arg_path_map, &fd_arg_task, &fd_arg_path, BPF_ANY);
-        }
+    if (submitEnter || submitSyscall) {
+        bpf_tail_call(ctx, &sys_enter_submit_tail, 0);
     }
 
+    // call syscall handler, if exists
+    bpf_tail_call(ctx, &sys_enter_tails, sys->id);
```

Comment:
This will never happen now, as the above tail call never returns.

Reply:
We have this tail call at the end of the submit tailcall if it's reached.